ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -60,6 +60,15 @@ azure_connection::list_objects(
     if (list_single)
         blob_parameters.PageSizeHint = 1;
 
+    try {
+        _azure_client.ListBlobs(blob_parameters);
+    } catch (const Azure::Core::RequestFailedException &e) {
+        return http_to_errno(e);
+    } catch (const std::exception &e) {
+        std::cerr << e.what() << std::endl;
+        return -1;
+    }
+
     auto list_blobs_response = _azure_client.ListBlobs(blob_parameters);
```

Comment:
could you include the `list_blobs_response` in the `try` block? To reduce code duplication.